TaskList/app/controllers/tasks_controller.rb

```diff
@@ -1,4 +1,12 @@
 class TasksController < ApplicationController
   def index
+    @title = "Task List"
+    @tasks = Task.all
+  end
+
+  def show
+    @title = "About This Task"
+    id = params[:id]
+    @tasks = Task.where(id: id)
```

Comment:
If you look at the difference between the `where` and the `find` in the SQL output, you'll see that `find` adds the `LIMIT 1` which will prevent you from retrieving more than one record - in error.